tests/cypress/tests/unit/lexeme_test.ts

```diff
@@ -0,0 +1,111 @@
+            expect(valueExpression.resolve(undefined, { value: '1E6' } as ISyntaxTree)).to.equal(1000000);
+            expect(valueExpression.resolve(undefined, { value: '0x100' } as ISyntaxTree)).to.equal(256);
+            expect(valueExpression.resolve(undefined, { value: '\\{abc' } as ISyntaxTree)).to.equal('{abc');
+            expect(valueExpression.resolve(undefined, { value: 'abc\\}' } as ISyntaxTree)).to.equal('abc}');
+            expect(valueExpression.resolve(undefined, { value: '\\}abc\\{' } as ISyntaxTree)).to.equal('}abc{');
+            expect(valueExpression.resolve(undefined, { value: '\\{\\{abc\\}\\}' } as ISyntaxTree)).to.equal('{{abc}}');
+
+            expect(valueExpression.resolve.bind(undefined, undefined, { value: '{abc' } as ISyntaxTree)).to.throw(Error);
+            expect(valueExpression.resolve.bind(undefined, undefined, { value: 'abc}' } as ISyntaxTree)).to.throw(Error);
+            expect(valueExpression.resolve.bind(undefined, undefined, { value: '}abc{' } as ISyntaxTree)).to.throw(Error);
+            expect(valueExpression.resolve.bind(undefined, undefined, { value: '{{abc}}' } as ISyntaxTree)).to.throw(Error);
+            expect(valueExpression.resolve.bind(undefined, undefined, { value: '"' } as ISyntaxTree)).to.throw(Error);
+            expect(valueExpression.resolve.bind(undefined, undefined, { value: '`' } as ISyntaxTree)).to.throw(Error);
+            expect(valueExpression.resolve.bind(undefined, undefined, { value: `'` } as ISyntaxTree)).to.throw(Error);
+            expect(valueExpression.resolve.bind(undefined, undefined, { value: '{' } as ISyntaxTree)).to.throw(Error);
+            expect(valueExpression.resolve.bind(undefined, undefined, { value: '}' } as ISyntaxTree)).to.throw(Error);
+        }
+    });
+});
+
+describe('operand', () => {
+    it('resolves values', () => {
+        expect(!!operand.resolve).to.equal(true);
+        expect(typeof operand.resolve).to.equal('function');
+
+        if (operand.resolve) {
+            expect(operand.resolve.bind(undefined, undefined, { value: 'abc' } as ISyntaxTree)).to.throw(Error);
+            expect(operand.resolve.bind(undefined, undefined, { value: '123' } as ISyntaxTree)).to.throw(Error);
+            expect(operand.resolve.bind(undefined, undefined, { value: '{abc' } as ISyntaxTree)).to.throw(Error);
+            expect(operand.resolve.bind(undefined, undefined, { value: 'abc}' } as ISyntaxTree)).to.throw(Error);
+            expect(operand.resolve.bind(undefined, undefined, { value: '{{abc}}' } as ISyntaxTree)).to.throw(Error);
+
+            expect(operand.resolve({ abc: 3 }, { value: '{abc}' } as ISyntaxTree)).to.equal(3);
+            expect(operand.resolve({ ['a bc']: 3 }, { value: '{a bc}' } as ISyntaxTree)).to.equal(3);
+            expect(operand.resolve({ ['{abc}']: 3 }, { value: '{\\{abc\\}}' } as ISyntaxTree)).to.equal(3);
+            expect(operand.resolve({ ['"abc"']: 3 }, { value: '{"abc"}' } as ISyntaxTree)).to.equal(3);
```

Comment:
how is `operand` different from `fieldExpression`? (and why are the keys here in `[...]`?)

Reply:
They differ only in terms of when they can be used in respect to their predecessors. Will clean up the implementation of the field expression a bit.

Reply:
`[...]` odd syntax for odd cases like property name generated through a function call

```
{ [myFunc()]: 'myValue' }
```

This is not really necessary here. Removing.